src/main/java/org/jboss/logmanager/handlers/PeriodicSizeRotatingFileHandler.java

```diff
@@ -156,7 +152,10 @@ public void setFile(final File file) throws FileNotFoundException {
             // Check for a rotate
             if (rotateOnBoot && maxBackupIndex > 0 && file != null && file.exists() && file.length() > 0L) {
                 try {
-                    rotate(file);
+                    final FileRotator fileRotator = getFileRotator();
+                    if (fileRotator != null) {
```

Comment:
This is likely never `null`. If anything it should always be a `FileRotator.EMPTY`.